### PR TITLE
[NIFI-12563] configure reporting task

### PR DESCRIPTION
**Pull request by scottyaslan**
[NIFI-12563](https://issues.apache.org/jira/browse/NIFI-12563)

Configure Reporting Task:

Kapture 2024-01-05 at 13 03 30

### CODE REVIEW

**Comment by rfellows** [comment]
reviewing...

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/reporting-tasks/reporting-tasks.reducer.ts [hunk 1]
```diff
@@ -61,6 +62,15 @@ export const reportingTasksReducer = createReducer(
...
+    on(configureReportingTaskSuccess, (state, { response }) => {
+        return produce(state, (draftState) => {
+            const componentIndex: number = draftState.reportingTasks.findIndex((f: any) => response.id === f.id);
+            if (componentIndex > -1) {
+                draftState.reportingTasks[componentIndex] = response.reportingTask;
+            }
+            draftState.saving = false;
+        });
+    }),
     on(createReportingTask, deleteReportingTask, (state, { request }) => ({
```
Comment (rfellows) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/reporting-tasks/reporting-tasks.reducer.ts:74]: We should add `configureReportingTask` here

```suggestion
    on(createReportingTask, deleteReportingTask, configureReportingTask, (state, { request }) => ({
        ...state,
        saving: true
    })),
```

**Review by rfellows — CHANGES_REQUESTED** [review]
A few things inline...


Also, the edit dialog's properties tab isn't the same width as the other 2 so when toggling between the tabs the dialog changed size. It isn't unique to this dialog. some of the other edit ones from the Controller settings do the same but we fixed this same issue with the parameters dialog. We should fix these as we find them.
Kapture 2024-01-08 at 10 40 42

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/reporting-tasks/reporting-tasks.reducer.ts (hunk 1 above)
Comment (scottyaslan) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/reporting-tasks/reporting-tasks.reducer.ts:74]: Good catch.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/reporting-tasks/edit-reporting-task/edit-reporting-task.component.html [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                        [createNewService]="createNewService"
+                        [getParameters]="getParameters"
+                        [parameterContext]="parameterContext"
+                        [goToParameter]="goToParameter"
+                        [convertToParameter]="convertToParameter"
+                        [goToService]="goToService"
+                        [supportsSensitiveDynamicProperties]="
+                            request.reportingTask.component.supportsSensitiveDynamicProperties
+                        ">
+                    </property-table>
```
[the remark this answers is not in the document]
Comment (mcgilman) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/reporting-tasks/edit-reporting-task/edit-reporting-task.component.html]: A JIRA was filed to track this issue: https://issues.apache.org/jira/browse/NIFI-12586

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/management-controller-services/index.ts [hunk 3]
```diff
@@ -16,6 +16,7 @@
  */
 
 import { Bundle, ControllerServiceEntity, Revision } from '../../../../state/shared';
+import { ReportingTaskEntity } from '../reporting-tasks';
```
Comment (rfellows) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/management-controller-services/index.ts:19]: This isn't used anymore and can be removed.

**Review by rfellows — APPROVED** [review]
Nice work, @scottyaslan. Will merge once CI completes

**Pull request merged**